src/components/common/button.tsx

```diff
@@ -0,0 +1,51 @@
+import React from 'react';
+
+interface ButtonProps {
+  text?: string;
+  onClick: () => void;
+  variant?: 'primary' | 'secondary' | 'tertiary';
+  size?: 'small' | 'medium' | 'large';
+  icon?: string;
+  disabled?: boolean;
+}
+
+export default function Button({
+  text,
+  onClick,
+  variant = 'primary',
+  size = 'medium',
+  icon,
+  disabled = false
+}: ButtonProps) {
+  // 모든 버튼에 기본으로 들어가는 스타일
+  const baseStyles = 'inline-flex items-center justify-center';
+
+  /* 버튼 스타일 디자인에 따라 추후 수정 */
+
+  // 버튼 스타일에 따라 지정
+  const VARIANT_STYLES = {
+    primary: 'bg-white text-gray-700',
+    secondary: 'bg-gray-100 text-black',
+    tertiary: 'bg-blue-400 text-white'
+  };
+
+  // 버튼 사이즈
+  const SIZE_STYLES = {
+    small: 'px-2 py-1 text-sm',
+    medium: 'px-4 py-2 text-base',
+    large: 'px-6 py-3 text-lg'
+  };
+
+  return (
+    <button
+      type="button"
+      onClick={onClick}
+      className={`${baseStyles} ${VARIANT_STYLES[variant]} ${SIZE_STYLES[size]} ${disabled ? 'opacity-50' : ''}`}
```

Comment:
```suggestion
      className={`${baseStyles} ${VARIANT_STYLES[variant]} ${SIZE_STYLES[size]} disabled:opacity-50`}
```
disabled를 가상 선택자로 선택할 수도 있습니다~